executor/executor.cc

```diff
@@ -732,30 +752,36 @@ void cover_enable(thread_t* th)
 	if (!flag_cover)
 		return;
 	debug("#%d: enabling /sys/kernel/debug/kcov\n", th->id);
-	if (ioctl(th->cover_fd, KCOV_ENABLE, 0)) {
-		// This should be fatal,
-		// but in practice ioctl fails with assorted errors (9, 14, 25),
-		// so we use exitf.
-		exitf("cover enable write failed");
+	if (flag_collect_comps) {
```

Comment:
I would suggest:
```
int kcov_mode = flag_collect_comps ? KCOV_MODE_TRACE_CMP : KCOV_MODE_TRACE_PC;
```
and them merge ioctl calls. Will be shorter and avoid duplication, in particular the comment after ioctl failure.